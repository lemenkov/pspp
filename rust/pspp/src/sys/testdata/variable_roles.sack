# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; # Layout code
7; # Nominal case size
0; # Not compressed
0; # Not weighted
0; # 1 case.
100.0; # Bias.
"01 Jan 11"; "20:53:52"; s64 "PSPP synthetic test file";
i8 0 *3;

# Variables.
2; 0; 0; 0; 0x050800 *2; s8 "I";
2; 0; 0; 0; 0x050800 *2; s8 "O";
2; 0; 0; 0; 0x050800 *2; s8 "B";
2; 0; 0; 0; 0x050800 *2; s8 "N";
2; 0; 0; 0; 0x050800 *2; s8 "P";
2; 0; 0; 0; 0x050800 *2; s8 "S";
2; 0; 0; 0; 0x050800 *2; s8 "X";

# Machine integer info record.
7; 3; 4; 8; 1; 2; 3; -1; 1; 1; ENDIAN; 1252;

# Variable attributes record.
7; 18; 1; COUNT (
"I:$@Role('0'"; i8 10; ")";
"/O:$@Role('1'"; i8 10; ")";
"/B:$@Role('2'"; i8 10; ")";
"/N:$@Role('3'"; i8 10; ")";
"/P:$@Role('4'"; i8 10; ")";
"/S:$@Role('5'"; i8 10; ")";
"/X:$@Role('6'"; i8 10; ")";
);

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;
