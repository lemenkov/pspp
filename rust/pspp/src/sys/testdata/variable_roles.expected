Unknown role "6".

╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
│       Version        │1.2.3                   │
├──────────────────────┼────────────────────────┤
│       Compression    │None                    │
│       Number of Cases│                       0│
╰──────────────────────┴────────────────────────╯

╭─────────┬────────────────────────╮
│Label    │PSPP synthetic test file│
│Variables│                       7│
╰─────────┴────────────────────────╯

                                               Variables
╭─┬────────┬─────┬─────────────────┬─────────┬─────┬─────────┬────────────┬────────────┬──────────────╮
│ │Position│Label│Measurement Level│   Role  │Width│Alignment│Print Format│Write Format│Missing Values│
├─┼────────┼─────┼─────────────────┼─────────┼─────┼─────────┼────────────┼────────────┼──────────────┤
│i│       1│     │                 │Input    │    8│Right    │F8.0        │F8.0        │              │
│o│       2│     │                 │Target   │    8│Right    │F8.0        │F8.0        │              │
│b│       3│     │                 │Both     │    8│Right    │F8.0        │F8.0        │              │
│n│       4│     │                 │None     │    8│Right    │F8.0        │F8.0        │              │
│p│       5│     │                 │Partition│    8│Right    │F8.0        │F8.0        │              │
│s│       6│     │                 │Split    │    8│Right    │F8.0        │F8.0        │              │
│x│       7│     │                 │Input    │    8│Right    │F8.0        │F8.0        │              │
╰─┴────────┴─────┴─────────────────┴─────────┴─────┴─────────┴────────────┴────────────┴──────────────╯

Data File and Variable Attributes
╭────────────────────────┬─────╮
│Variable and Name       │Value│
├────────────────────────┼─────┤
│i                 $@Role│0    │
├────────────────────────┼─────┤
│o                 $@Role│1    │
├────────────────────────┼─────┤
│b                 $@Role│2    │
├────────────────────────┼─────┤
│n                 $@Role│3    │
├────────────────────────┼─────┤
│p                 $@Role│4    │
├────────────────────────┼─────┤
│s                 $@Role│5    │
├────────────────────────┼─────┤
│x                 $@Role│6    │
╰────────────────────────┴─────╯
